File: Python/Sliding Window/Maximum Erasure Value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        left=0
        curr_sum=0
        max_sum=0
        seenSet=set()
        for right in range(0,len(nums)):
            while nums[right] in seenSet:
                seenSet.remove(nums[left])
                curr_sum-=nums[left]
                left+=1
            seenSet.add(nums[right])
            curr_sum+=nums[right]
            max_sum=max(max_sum,curr_sum)
        return max_sum
        

File: Java/Sliding Window/Maximum Erasure Value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        int curr_sum=0;
        int max_sum=Integer.MIN_VALUE;
        int left=0;
        HashSet<Integer> seen=new HashSet<>();
        for(int right=0;right<nums.length;right++){
            while(seen.contains(nums[right])){
                curr_sum-=nums[left];
                seen.remove(nums[left]);
                left++;
            }
            seen.add(nums[right]);
            curr_sum+=nums[right];
            max_sum=Math.max(max_sum,curr_sum);
        }
        return max_sum;
    }
}
